berbix.go

```diff
@@ -122,11 +141,106 @@ func (c *defaultClient) UpdateTransaction(tokens *Tokens, options *UpdateTransac
 	}
 
 	metadata := &TransactionMetadata{}
-	return metadata, c.tokenAuthRequest(http.MethodPatch, tokens, v0Transactions, options, metadata)
+	return metadata, c.tokenAuthRequestExpecting2XX(http.MethodPatch, tokens, v0Transactions, options, metadata)
 }
 
 func (c *defaultClient) OverrideTransaction(tokens *Tokens, options *OverrideTransactionOptions) error {
-	return c.tokenAuthRequest(http.MethodPatch, tokens, "/v0/transactions/override", options, nil)
+	return c.tokenAuthRequestExpecting2XX(http.MethodPatch, tokens, "/v0/transactions/override", options, nil)
+}
+
+// UploadImage uploads an image to Berbix and provides a response that indicates the next upload, if any,
+// that is expected, along with flags indicating any issues with the image that could immediately be detected.
+// UploadImage expects a slice of bytes representing an image in a supported format, such as JPEG or PNG.
+// Returns a InvalidStateErr if the upload was invalid for the current state of the transaction.
+// Returns a TransactionDoesNotExistErr if the transaction for which the image is being uploaded no longer exists.
+// Returns a PayloadTooLargeErr if the uploaded payload or underlying image is too large.
+func (c *defaultClient) UploadImage(image []byte, subject ImageSubject, format ImageFormat, tokens *Tokens) (*ImageUploadResult, error) {
```

Comment:
we probably aren't going to need the imageFormat much longer, not sure if we want to wait to omit it or not.

Reply:
That's a good point -- I switched to following the pattern of accepting an `*Options` parameter so that we can later mark the `ImageFormat` struct field as deprecated and say that it can be left unset if we no longer need it.